### PR TITLE
Reduce metric-related code duplication in ChartsHolder, LocationPageHeader, and LocationHeaderStats

### PR DESCRIPTION
**Pull request by mikelehen**
There were a few places where we repeated very similar code 5 times, once for each metric, and I simplified it to just map over ALL_METRICS.

### CODE REVIEW

File: src/components/LocationPage/ChartBlock.tsx [hunk 1]
```diff
@@ -25,44 +25,38 @@ function ChartBlock(props: {
   shareButtonProps: { [key: string]: any };
   metric: Metric;
   projections: Projections;
-  data?: any;
```
Comment (mikelehen) [inline src/components/LocationPage/ChartBlock.tsx:28]: I did some minor refactoring (elevated projections, metric, isMobile to local variables) but the only major change here is removing `data` and instead relying on `projections.hasMetric(metric)` to decide if there is data.

File: src/components/LocationPage/ChartBlock.tsx [hunk 2]
```diff
@@ -25,44 +25,38 @@ function ChartBlock(props: {
...
   stateId: string;
-  countyId?: string;
```
Comment (mikelehen) [inline src/components/LocationPage/ChartBlock.tsx:30]: unused

File: src/components/SummaryStats/LocationHeaderStats.tsx [hunk 3]
```diff
@@ -61,7 +61,6 @@ const SummaryStat = ({
   onClick,
   beta,
   condensed,
-  flipSignalStatusOrder,
```
Comment (mikelehen) [inline src/components/SummaryStats/LocationHeaderStats.tsx:64]: unused

**Review by chasulin — APPROVED** [review]
Sweet!

**Pull request merged**